src/UI/InGameMenu.java

```diff
@@ -23,20 +24,24 @@
 
   public InGameMenu(Collection<T> options)
   {
-    menuOptions = new ArrayList<T>();
-    menuOptions.addAll(options);
-    optionSelector = new OptionSelector(menuOptions.size());
-    wasReset = true;
+    this(options, new OptionSelector(options.size()));
   }
 
   public InGameMenu(T[] options)
+  {
+    this(new ArrayList<T>(Arrays.asList(options)));
+  }
+
+  public InGameMenu(T[] options, OptionSelector selector)
+  {
+    this(new ArrayList<T>(Arrays.asList(options)), selector);
+  }
+
+  public InGameMenu(Collection<T> options, OptionSelector selector)
   {
     menuOptions = new ArrayList<T>();
-    for( int i = 0; i < options.length; ++i )
-    {
-      menuOptions.add( options[i] );
-    }
-    optionSelector = new OptionSelector(menuOptions.size());
+    menuOptions.addAll(options);
+    optionSelector = selector;
```

Comment:
I feel like we should at least verify `options.size() == selector.numOptions`. That would require adding an accessor to `OptionSelector`, but it's at least a sensible one.

Reply:
And throw an exception if the check doesn't pass?

Reply:
mmm, yeah. I think that's probably best; probably with an unchecked exception type. IllegalArgumentException strikes me as a likely candidate, but do whatever you think makes sense.